src/Query/Filter/PriorityField.ts

```diff
@@ -39,6 +39,8 @@ export class PriorityField extends Field {
                 filter = (task: Task) => (task.priority ? task.priority.localeCompare(filterPriority!) < 0 : false);
             } else if (priorityMatch[2] === 'below') {
                 filter = (task: Task) => (task.priority ? task.priority.localeCompare(filterPriority!) > 0 : false);
+            } else if (priorityMatch[2] === 'not') {
+                filter = (task: Task) => (task.priority ? task.priority !== filterPriority : false);
```

Comment:
I opted for line 43 instead of the simpler:

```ts
                filter = (task: Task) => task.priority === filterPriority;
```

All tests pass with the simpler version, but I felt consistency was probably good.

As far as I can see, task.priority can never be null, so I wondered about getting rid of the ternary operator on all 4 filters.

Would appreciate any feedback.

Reply:
Yes, I would remove that check for task.priority everywhere since the priority is a non-null property of the task.

Maybe you can replace the "else ifs" with a "switch" statement.

Reply:
Great. Thank you.

Reply:
> Yes, I would remove that check for task.priority everywhere since the priority is a non-null property of the task.
> 
> Maybe you can replace the "else ifs" with a "switch" statement.

Thanks. I've done that in two stages. here's the second one, showing the final code.

https://github.com/obsidian-tasks-group/obsidian-tasks/pull/1365/commits/7c4dd7203765c801f0af8d08b90275338b5ff556